source/Calamari.Scripting/Behaviours/ThrowScriptErrorIfNeededBehaviour.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Threading.Tasks;
+using Calamari.Common.Commands;
+using Calamari.Common.Plumbing.Pipeline;
+using Calamari.Scripting;
+
+namespace Calamari.Commands
+{
+    class ThrowScriptErrorIfNeededBehaviour : IOnFinishBehaviour
+    {
+        public bool IsEnabled(RunningDeployment context)
+        {
+            return true;
+        }
+
+        public Task Execute(RunningDeployment context)
+        {
+            var exitCode = context.Variables.GetInt32(SpecialVariables.Action.Script.ExitCode);
+            if (exitCode != 0)
+            {
+                throw new CommandException($"Script returned non-zero exit code: {exitCode}.");
+            }
+            return this.CompletedTask();
```

Comment:
Do we need to handle `FailScriptOnErrorOutput` ?
https://github.com/OctopusDeploy/Calamari/blob/0a04064910e32b9c7c2bc519c3ea9d08ded9d3af/source/Calamari/Kubernetes/Conventions/HelmUpgradeConvention.cs#L53

https://github.com/OctopusDeploy/Calamari/blob/8ceade5538f889b3d2d237811a73b8cd602d76aa/source/Calamari/Deployment/Conventions/ExecuteScriptConvention.cs#L30